### PR TITLE
Adapt Java Component changes in API Version 20240202preview

### PR DESCRIPTION
**Pull request by Caoxuyang**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/containerapp/HISTORY.rst [hunk 1]
```diff
@@ -13,6 +13,9 @@ upcoming
 * 'az containerapp update': Fix --scale-rule-tcp-concurrency for TCP scale rule
 * 'az containerapp compose create': Fix an issue where the environment's location is not resolved from --location
 * 'az containerapp up': Fix an issue about creating resource group automatically
+* 'az containerapp env java-component admin-for-spring': Support create/update/show/delete Admin for spring
+* 'az containerapp env java-component nacos': Support create/update/show/delete Nacos
+* 'az containerapp env java-component': Support --service-binds, --unbind_service_bindings
```
Comment (ShichaoQiu) [inline src/containerapp/HISTORY.rst:18]: typo, pls check the parameter name

File: src/containerapp/azext_containerapp/_params.py [hunk 2]
```diff
@@ -343,7 +343,9 @@ def load_arguments(self, _):
         c.argument('java_component_name', options_list=['--name', '-n'], help="The Java component name.")
         c.argument('environment_name', options_list=['--environment'], help="The environment name.")
         c.argument('resource_group_name', arg_type=resource_group_name_type, id_part=None)
-        c.argument('configuration', nargs="*", help="Java component configuration. Configuration must be in format \"<propertyName>=<value> <propertyName>=<value> ...\".")
+        c.argument('service_bindings', nargs='*', options_list=['--bind'], help="Space separated list of services, bindings or other Java components to be connected to this Java Component. e.g. SVC_NAME1[:BIND_NAME1] SVC_NAME2[:BIND_NAME2]...")
+        c.argument('unbind_service_bindings', nargs='*', options_list=['--unbind'], help="Space separated list of services, bindings or Java components to be removed from this app. e.g. BIND_NAME1...")
```
Comment (ShichaoQiu) [inline src/containerapp/azext_containerapp/_params.py:347]: change ` components to be removed from this app.` to ` components to be removed from this Java Component.`?

File: src/containerapp/azext_containerapp/java_component_decorator.py [hunk 3]
```diff
@@ -28,15 +32,32 @@ def get_argument_configuration(self):
 
     def get_argument_environment_name(self):
         return self.get_param("environment_name")
+    
+    def get_environment_id(self, cmd: AzCliCommand):
```
Comment (ShichaoQiu) [inline src/containerapp/azext_containerapp/java_component_decorator.py:36]: It seems that it is a common logic, is there any existing method that we can reuse?

File: src/containerapp/azext_containerapp/_params.py (hunk 2 above)
Comment (Caoxuyang) [reply src/containerapp/azext_containerapp/_params.py]: changed

File: src/containerapp/HISTORY.rst (hunk 1 above)
Comment (Caoxuyang) [reply src/containerapp/HISTORY.rst]: changed

File: src/containerapp/azext_containerapp/java_component_decorator.py (hunk 3 above)
Comment (Caoxuyang) [reply src/containerapp/azext_containerapp/java_component_decorator.py:36]: I didn't find one. I will take another look.

**Comment by bowen5** [comment]
tagging @Greedygre

**Review by ShichaoQiu — APPROVED** [review]

File: src/containerapp/azext_containerapp/commands.py [hunk 4]
```diff
@@ -205,6 +205,18 @@ def load_command_table(self, args):
         g.custom_show_command('show', 'show_eureka_server_for_spring')
         g.custom_command('delete', 'delete_eureka_server_for_spring', confirmation=True, supports_no_wait=True)
 
+    with self.command_group('containerapp env java-component nacos') as g:
+        g.custom_command('create', 'create_nacos', supports_no_wait=True)
+        g.custom_command('update', 'update_nacos', supports_no_wait=True)
+        g.custom_show_command('show', 'show_nacos')
+        g.custom_command('delete', 'delete_nacos', confirmation=True, supports_no_wait=True)
+
+    with self.command_group('containerapp env java-component admin-for-spring') as g:
```
Comment (Juliehzl) [inline src/containerapp/azext_containerapp/commands.py:214]: add is_preview for new command group

Comment (Caoxuyang) [reply src/containerapp/azext_containerapp/commands.py:214]: Line 179 added is_preview for containerap env java-component command group.

**Review by Moazzem-Hossain-pixel — CHANGES_REQUESTED** [review]
Pls bug fixes

**Review by Moazzem-Hossain-pixel — APPROVED** [review]
Thanks

**Review by Juliehzl — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**